cli/commands/site/set-domain.ts

```diff
@@ -0,0 +1,104 @@
+import { __, _n } from '@wordpress/i18n';
+import { getDomainNameValidationError } from 'common/lib/domains';
+import { arePathsEqual } from 'common/lib/fs-utils';
+import { SiteCommandLoggerAction as LoggerAction } from 'common/logger-actions';
+import {
+	getSiteByFolder,
+	lockAppdata,
+	readAppdata,
+	saveAppdata,
+	unlockAppdata,
+} from 'cli/lib/appdata';
+import { removeDomainFromHosts } from 'cli/lib/hosts-file';
+import { connect, disconnect } from 'cli/lib/pm2-manager';
+import { setupCustomDomain } from 'cli/lib/site-utils';
+import {
+	isServerRunning,
+	startWordPressServer,
+	stopWordPressServer,
+} from 'cli/lib/wordpress-server-manager';
+import { Logger, LoggerError } from 'cli/logger';
+import { StudioArgv } from 'cli/types';
+
+export async function runCommand( sitePath: string, domainName: string ): Promise< void > {
+	const logger = new Logger< LoggerAction >();
+
+	try {
+		logger.reportStart( LoggerAction.LOAD_SITES, __( 'Loading site…' ) );
+		const site = await getSiteByFolder( sitePath, false );
+		logger.reportSuccess( __( 'Site loaded' ) );
+
+		try {
+			await lockAppdata();
+			const appdata = await readAppdata();
+
+			const existingDomains = appdata.sites
+				.map( ( site ) => site.customDomain )
+				.filter( ( domain ): domain is string => Boolean( domain ) );
+			const domainError = getDomainNameValidationError( true, domainName, existingDomains );
+
+			if ( domainError ) {
+				throw new LoggerError( domainError );
+			}
+
+			const site = appdata.sites.find( ( site ) => arePathsEqual( site.path, sitePath ) );
+			if ( ! site ) {
+				throw new LoggerError( __( 'The specified folder is not added to Studio.' ) );
+			}
```

Comment:
Do we still need this check after `getSiteByFolder`?

Reply:
Good call 👍